apps/klasses/models.py

```diff
@@ -81,3 +83,27 @@ def save(self, *args, **kwargs):
             self.created = timezone.now()
         self.modified = timezone.now()
         return super().save(*args, **kwargs)
+
+    def homeworks_completely_graded(self):
+        for hw in self.homework_definitions.all():
+            if hw.team_based:
+                for team in self.teams.all():
+                    team_graded = False
+                    for submission in Submission.objects.filter(team=team, definition=hw):
```

Comment:
More of a nitpick/suggestion, but I think we could probably code golf this down to a double list comprehension and then leveraging `all()` here potentially... It may be a little gross, but something we can take a look at?

Reply:
I think that this could be done with list comprehensions, but this function doesn't actually quite do what I wanted it to do. I fixed it on the `consolidation` branch. After that fix, the function may get a little unwieldy as a list comprehension. Check it out and lmk what you think. 

Additionally, we probably should continue the review on that branch, since it includes all of the recent changes of `forms`.